lib/min_heap/min_heap.c

```diff
@@ -71,13 +71,14 @@ static void heapify_down(struct min_heap *heap, size_t index)
 {
 	size_t left, right, smallest;
 
-	while (true) {
-		left = 2 * index + 1;
-		right = 2 * index + 2;
+	left = 2 * index + 1;
+
+	/* Terminate loop when the left child is out of bounds */
+	for (; left < heap->size; left = 2 * index + 1) {
+		right = left + 1;
 		smallest = index;
```

Comment:
All of the locals could be bound to the loop as wel.

```suggestion

	/* Terminate loop when the left child is out of bounds */
	for (size_t left = 2 * index + 1; left < heap->size; left = 2 * index + 1) {
		size_t right = left + 1;
 		size_t smallest = index;
```